Patch contains 2 changes:
- enable readline on FreeBSD (upstream never accepted this PR)
- use -O3 on FreeBSD

--- Makefile.orig	2021-04-13 02:22:04 UTC
+++ Makefile
@@ -17,7 +17,8 @@ endif
 
 # Compiler flags for various configurations:
 
-CFLAGS := -std=c99 -pedantic -Wall -Wextra -Wno-unused-parameter
+# change to c11 on FreeBSD because FreeBSD headers induce c11 through isnan(3), etc.
+CFLAGS := -std=c11 -pedantic -Wall -Wextra -Wno-unused-parameter
 
 ifeq "$(CC)" "clang"
   CFLAGS += -Wunreachable-code
@@ -27,13 +28,18 @@ ifeq "$(shell uname)" "Linux"
   HAVE_READLINE := yes
 endif
 
+ifeq "$(shell uname)" "FreeBSD"
+  HAVE_READLINE := yes
+  CFLAGS += -I$(prefix)/include
+endif
+
 ifeq "$(build)" "debug"
   CFLAGS += -g
 else ifeq "$(build)" "sanitize"
   CFLAGS += -pipe -g -fsanitize=address -fno-omit-frame-pointer
   LDFLAGS += -fsanitize=address
 else ifeq "$(build)" "release"
-  CFLAGS += -O2
+  CFLAGS += -O3
   LDFLAGS += -Wl,-s
 endif
 
